mbuild/compound.py

```diff
@@ -358,17 +358,22 @@ def mass(self):
         if self._contains_only_ports():
             return self._particle_mass(self)
         else:
+            particle_masses = [self._particle_mass(p) for p in self.particles()]
+            if None in particle_masses:
+                warn(
+                    f"Some particle in this Compound has not had their mass set."
+                )
             return sum([self._particle_mass(p) for p in self.particles()])
```

Comment:
I don't think this `sum` operation will work if `particle_masses` contains `None` values.  

Since for a compound containing children, this `mass` function is returning a summation of all particle masses, we either have to make an assumption about the mass of particles with `particle.mass == None`. For example, we consider them zero for the purposes of this summation, but still give a warning. Or, we decide that a summation of particle masses only works if every particle has had their mass defined, and in that case returns an Error, or maybe it just returns `None` with the warning.